pkgs/games/multimc/default.nix

```
@@ -28,14 +28,30 @@ in mkDerivation rec {
       --replace 'QString MSAClientID = "";' 'QString MSAClientID = "${msaClientID}";'
   '';
 
-  cmakeFlags = [ "-DMultiMC_LAYOUT=lin-system" ];
+  cmakeFlags = [ "-DLauncher_LAYOUT=lin-nodeps" ];
+
+  desktopItem = makeDesktopItem {
+    name = "multimc";
+    desktopName = "MultiMC";
+    genericName = "Minecraft Launcher";
+    comment = "Free, open source launcher and instance manager for Minecraft.";
+    icon = "launcher";
+    exec = "DevLauncher";
+    categories = "Game";
+    terminal = "false";
+  };
+
+  preFixup = ''
+    mkdir -p $out/lib
+    mv $out/bin/*.so $out/lib/
+  '';
 
   postInstall = ''
-    install -Dm644 ../launcher/resources/multimc/scalable/multimc.svg $out/share/pixmaps/multimc.svg
-    install -Dm755 ../launcher/package/linux/multimc.desktop $out/share/applications/multimc.desktop
+    install -Dm644 ../launcher/resources/multimc/scalable/launcher.svg $out/share/pixmaps/launcher.svg
+    install -Dm755 ${desktopItem}/share/applications/multimc.desktop $out/share/applications/multimc.desktop
 
     # xorg.xrandr needed for LWJGL [2.9.2, 3) https://github.com/LWJGL/lwjgl/issues/128
-    wrapProgram $out/bin/multimc \
+    wrapProgram $out/bin/DevLauncher \
       --set GAME_LIBRARY_PATH /run/opengl-driver/lib:${libpath} \
       --prefix PATH : ${lib.makeBinPath [ xorg.xrandr ]}
```

```suggestion
      --prefix PATH : ${lib.makeBinPath [ xorg.xrandr ]}```suggestion
      --prefix PATH : ${lib.makeBinPath [ xorg.xrandr ]}
      --add-flags '-d "''${XDG_DATA_HOME-$HOME/.local/.share}/multimc"'
```
```

---

As suggested https://github.com/MultiMC/Launcher/issues/4328#issuecomment-991923874, you can use this flag to specify where to store data. Otherwise, it would attempt to store it in the same directory as the application